### PR TITLE
feat(source-mongodb-v2): support sharded clusters and object-type _id, bump Debezium and base image

### PR DESCRIPTION
**Pull request by rodireich**
## What
Unblocks MongoDB sharded clusters, adds support for `_id` values that are embedded documents, and moves the connector to Debezium 3.x.

- `check` no longer fails when the cluster type is not `REPLICA_SET`. It logs a warning instead, so sharded clusters connected through `mongos` can be used.
- New `IdType.OBJECT` (BSON `object`): `_id` values that are documents are stored in state as extended JSON and parsed back when resuming the initial snapshot.
- The pre-flight `_id` type check now uses a constant number of `_id`-index seeks instead of a `$group`-by-`$type` aggregate that scanned the whole collection (from #84962).
- Debezium `2.6.2.Final` -> `3.6.2.Final` (pulls in Mongo driver 5.9.2 and Kafka Connect 4.x). Base image bumped to `java-connector-base:2.0.4`. Connector version `2.0.7` -> `2.1.0`.

## How
- `MongoDbSource.check`: warn instead of returning `FAILED` for non-replica-set cluster types.
- `IdType` / `MongoDbInitialLoadRecordIterator`: add `OBJECT`, serialize with `JsonMode.EXTENDED`, parse back with `BsonDocument.parse`.
- `InitialSnapshotHandler.getIdFieldTypes`: two covered min/max seeks on the `_id` index find the endpoint BSON types; if they share a single-type bracket the collection is homogeneous, otherwise each candidate type is probed with one index-bounded `$type` query. The existing multiple-type warning and unsupported-type `ConfigErrorException` are unchanged, and `object` is now an accepted `_id` type.
- `MongoDbDebeziumStateUtil`: Kafka Connect 4.x no longer defaults `bootstrap.servers` and rejects the file offset store config without it. Supply a placeholder, as Debezium's own embedded engine does.
- Debezium 3.x compatibility for CDC: `snapshot.mode` is now `no_data` (`never` was removed in Debezium 3). Debezium 3 also stores resume tokens in offsets as base64-encoded BSON instead of the hex `_data` string; the connector now accepts both forms wherever it reads an offset and normalizes the token back to the hex form when saving state, so persisted state keeps its previous shape.
- Tests: assert sharded clusters pass `check`, make driver-name assertions order-insensitive for driver 5.x, add `getIdFieldTypes` tests (empty, single type, interleaved numeric bracket, mixed brackets), and mock the new index seeks in `MongoDbCdcInitializerTest`.

## Review guide
1. `MongoDbSource.java` - check behavior
2. `IdType.java`, `MongoDbInitialLoadRecordIterator.java` - object-type `_id` support
3. `InitialSnapshotHandler.java` - index-seek `_id` type detection
4. `MongoDbDebeziumStateUtil.java`, `build.gradle` - Debezium 3.x

## User Impact
- Sharded clusters pass `check` and can be synced with CDC.
- Collections whose `_id` is an embedded document are supported.
- The `_id` type check before the initial snapshot no longer scans the whole collection, so large collections start syncing much sooner.
- Saved CDC state format is unchanged; existing connections resume from their saved resume tokens.

## Can this PR be safely reverted and rolled back?
- [x] YES 💚 (connections that started relying on sharded clusters or object-type `_id` would fail after a revert)
- [ ] NO ❌

> [!IMPORTANT]
> Active progressive rollout warning for source-postgres.

- [ ] (Click to Approve:) Bypass the active progressive rollout warning for source-postgres in the PR comment [here](https://github.com/airbytehq/airbyte/pull/83705#issuecomment-5277408699).

---

> [!IMPORTANT]
> **Autopilot Progressive Rollout Enabled**
>
> [Autopilot progressive rollouts](https://github.com/airbytehq/airbyte-ops-mcp/blob/main/docs/autopilot-rollouts.md) are enabled for one or more connector(s) modified in this PR. Check the box below if you need to bypass normal rollout safety processes and release to all users immediately upon merge:
>
> - [ ] ⚡ **Release immediately** (bypasses automatic progressive rollout)
>
> **Note:**
>
> - ⚠️ The above bypass option is for emergency/hotfix use only.
> - 🔗 You can monitor or manually advance a rollout at **[ops.internal.airbyte.ai](https://ops.internal.airbyte.ai)**.

### CODE REVIEW

**Comment by rodireich** [comment]
/format-fix
> Format-fix job started... [Check job output.][1]

[1]: https://github.com/airbytehq/airbyte/actions/runs/33684314823
> ✅ Changes applied successfully. (c9c5f3ccbd8a8a04818b69c26dd58338bb0d7b7f)

File: airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoDbSource.java [hunk 1]
```diff
@@ -98,10 +98,7 @@ public AirbyteConnectionStatus check(final JsonNode config) {
         }
 
         if (!ClusterType.REPLICA_SET.equals(mongoClient.getClusterDescription().getType())) {
-          LOGGER.error("Target MongoDB instance is not a replica set cluster.");
-          return new AirbyteConnectionStatus()
-              .withMessage("Target MongoDB instance is not a replica set cluster.")
-              .withStatus(AirbyteConnectionStatus.Status.FAILED);
+          LOGGER.warn("Target MongoDB instance is not a replica set cluster.{}", mongoClient.getClusterDescription().getType());
         }
```
[the remark this answers is not in the document]
Comment (rodireich) [reply airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoDbSource.java]: Addressed in 000ea13: the cluster type is captured once and the message is now `MongoDB instance is not a replica set cluster (cluster type: {}).` `MongoDbSourceTest` now covers the sharded case.

File: airbyte-integrations/connectors/source-mongodb-v2/build.gradle [hunk 2]
```diff
@@ -53,10 +53,10 @@ dependencies {
     dataGeneratorImplementation ('com.github.javafaker:javafaker:1.0.2') { exclude module: 'snakeyaml' }
     dataGeneratorImplementation 'io.github.oshai:kotlin-logging-jvm:5.1.0'
     dataGeneratorImplementation 'org.jetbrains.kotlinx:kotlinx-cli-jvm:0.3.5'
-    dataGeneratorImplementation 'org.mongodb:mongodb-driver-sync:4.10.2'
+//    dataGeneratorImplementation 'org.mongodb:mongodb-driver-sync:4.10.2'
 
-    debeziumTestImplementation 'io.debezium:debezium-embedded:2.6.0.Final'
-    debeziumTestImplementation 'io.debezium:debezium-connector-mongodb:2.6.0.Final'
+    debeziumTestImplementation 'io.debezium:debezium-embedded:3.6.0.Final'
+    debeziumTestImplementation 'io.debezium:debezium-connector-mongodb:3.6.0.Final'
```
[the remark this answers is not in the document]
Comment (rodireich) [reply airbyte-integrations/connectors/source-mongodb-v2/build.gradle]: Addressed in 000ea13: the main `implementation` dependencies are bumped to `3.6.2.Final` (latest 3.x Final) and `debeziumTest` is aligned to the same version. Kafka Connect 4.x, which Debezium 3.x pulls in, requires `bootstrap.servers` when configuring the file offset store, so `MongoDbDebeziumStateUtil` now supplies a placeholder the same way Debezium's embedded engine does.

File: airbyte-integrations/connectors/source-postgres/gradle.properties [hunk 3]
```diff
@@ -1,3 +1,3 @@
 testExecutionConcurrency=-1
 JunitMethodExecutionTimeout=5m
-cdkVersion=1.1.8
+cdkVersion=local
```
[the remark this answers is not in the document]
Comment (rodireich) [reply airbyte-integrations/connectors/source-postgres/gradle.properties]: Reverted in 000ea13: `cdkVersion` is back to `1.1.8`. The change was a local-development leftover.

**Comment by rodireich** [comment]
/format-fix
> Format-fix job started... [Check job output.][1]

[1]: https://github.com/airbytehq/airbyte/actions/runs/33685511262
> 🟦 Job completed successfully (no changes).

**Comment by rodireich** [comment]
/format-fix
> Format-fix job started... [Check job output.][1]

[1]: https://github.com/airbytehq/airbyte/actions/runs/33690657799
> ✅ Changes applied successfully. (8c8fabf9e59037e5b5201d11f00495ef37c74ec7)
